oauth2_lib/fastapi.py

```diff
@@ -367,3 +367,70 @@ async def _opa_decision(
         return None
 
     return _opa_decision
+
+
+def opa_graphql_decision(
+    opa_url: str,
+    _oidc_security: OIDCUser,
+    enabled: bool = True,
+    auto_error: bool = True,
+    opa_kwargs: Mapping[str, str] | None = None,
+) -> Callable[[str], Coroutine[Any, Any, bool | None]]:
+    async def _opa_decision(
+        path: str,
+        oidc_user: OIDCUserModel,
+    ) -> bool | None:
+        """
+        Check OIDCUserModel against the OPA policy.
+
+        This is used as a security module in Graphql projects
+        This method will make an async call towards the Policy agent.
```

Comment:
It might be nice to elaborate a bit on how to use it. E.g. describe some use cases/examples. It's now very abstract (IMHO).

Reply:
I think that is a good idea. Should probably go into a separate docs folder. Will postpone this for now since we are in a bit of hurry to use this and there will probably be some changes and additional code that is now still in Pythia and will be moved to this lib.